File: LeetCode/Graph/BFS/Golang/Leetcode515-find-largest-value-in-each-tree-row.go
package sbgo

func largestValues(root *TreeNode) []int {
	ans := []int{}
	if root == nil {
		return ans
	}
	q := []*TreeNode{root}
	for len(q) != 0 {
		n, cur := len(q), int(-3e9)
		for i := 0; i < n; i++ {
			var t = q[0]
			q = q[1:]
			cur = max(cur, t.Val)
			if t.Left != nil {
				q = append(q, t.Left)
			}
			if t.Right != nil {
				q = append(q, t.Right)
			}
		}
		ans = append(ans, cur)
	}
	return ans
}

File: LeetCode/Graph/BFS/Python/Leetcode515-find-largest-value-in-each-tree-row.py
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        if not root: return ans
        q = deque()
        q.append(root)
        while q:
            cur, n = int(-2e39), len(q)
            for i in range(n):
                t = q.popleft()
                cur = max(cur, t.val)
                if t.left: q.append(t.left)
                if t.right: q.append(t.right)
            ans.append(cur)
        return ans
